resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -123,6 +123,10 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
     checkResult(finalState, result)
   }
 
+  test("yarn-cluster should use NM's public address instead  of SPARK_LOCAL_*") {
+    testBasicYarnApp(false, conf = Map("spark.yarn.appMasterEnv.SPARK_LOCAL_IP" -> "1.1.1.1"))
```

Comment:
1.1.1.1 is a valid IP address (try it out, ping it!); you should probably use something invalid here instead (like "0.1.1.1", which you shouldn't be able to use as a target IP address).

Reply:
They are both valid addresses. `ping`-ability is orthogonal. the point was that you can't bind to it.
```
$ nc -l 1.1.1.1 4040
nc: Can't assign requested address
```
your comment inspired me though to use 0.0.0.1 which can be used only as a [source address](https://tools.ietf.org/rfc/rfc3330.txt) to signify the issue .